### PR TITLE
[SW-1599] Target Encoder Tests Covering Various Order of Columns

### PR DESCRIPTION
**Pull request by mn-mikke**

### CODE REVIEW

File: ml/src/test/scala/ai/h2o/sparkling/ml/features/H2OTargetEncoderTestSuite.scala [hunk 1]
```diff
@@ -322,4 +322,130 @@ class H2OTargetEncoderTestSuite extends FunSuite with Matchers with SharedH2OTes
...
+    }
+
+    val expectedResult = transformTrainingDataset(Array("RACE", "DPROS", "DCAPS"))
+    val result = transformTrainingDataset(Array("DPROS", "DCAPS", "RACE"))
+      .select('ID, 'CAPSULE, 'AGE , 'RACE, 'DPROS, 'DCAPS, 'PSA, 'VOL, 'GLEASON, 'RACE_te, 'DPROS_te, 'DCAPS_te)
+
+    TestFrameUtils.assertDataFramesAreIdentical(expectedResult, result)
+  }
+
+  test("TargetEncoderMOJOModel returns the same result regardless the order of the inputCols specification") {
```
Comment (mn-mikke) [inline ml/src/test/scala/ai/h2o/sparkling/ml/features/H2OTargetEncoderTestSuite.scala]: This test fails ...

File: ml/src/test/scala/ai/h2o/sparkling/ml/features/H2OTargetEncoderTestSuite.scala [hunk 2]
```diff
@@ -322,4 +322,130 @@ class H2OTargetEncoderTestSuite extends FunSuite with Matchers with SharedH2OTes
 
     TestFrameUtils.assertDataFramesAreIdentical(transformedByModel, transformedByMOJOModel)
   }
+
+  test("TargetEncoderModel returns the same result regardless the order of the inputCols specification") {
```
Comment (jakubhava) [inline ml/src/test/scala/ai/h2o/sparkling/ml/features/H2OTargetEncoderTestSuite.scala]: Should we mark this test as skipped until it is fixed in H2O?

Comment (mn-mikke) [reply ml/src/test/scala/ai/h2o/sparkling/ml/features/H2OTargetEncoderTestSuite.scala]: Will do thanks!

**Comment by Pscheidl** [comment]
One of the tests is failing according to @mn-mikke , which means merging this would be a problem ?

The related issue is: https://0xdata.atlassian.net/browse/PUBDEV-6846

**Comment by mn-mikke** [comment]
Thanks @Pscheidl, created the ticket [SW-1602](https://0xdata.atlassian.net/browse/SW-1602) to enable the test after [PUBDEV-6846](https://0xdata.atlassian.net/browse/PUBDEV-6846) is fixed.

**Pull request merged**